libsrc/common/getcpu.s

```diff
@@ -27,20 +29,39 @@ _getcpu:
         cmp     #1
         bcc     @L9
 
-; This is at least a 65C02, check for a 4510
+; This is at least a 65C02, check for a 65ce02/4510
 
-        .byte   $42,$ea         ; neg on 4510, nop #$ea on 65c02, wdm $ea on 65816
+        .byte   $42,$ea         ; neg on 65ce02/4510, nop #$ea on 65c02, wdm $ea on 65816
         cmp     #1
-        bne     @L8
+        beq     @L6
+
+; check for 4510
+
+        lda     #5              ; CPU_65CE02 constant
+        .byte   $5c             ; map on 4510, aug on 65ce02 (acts like 4 byte nop)
+        lda     #3              ; CPU_4510 constant
+        nop
+        bne     @L9
+
+; check for 65sc02
+
+@L6:    ldy     $f7
+        ldx     #$00
+        stx     $f7
+        .byte   $f7,$f7         ; nop nop on 65sc02, smb7 $f7 on 65c02 and 65816
```

Comment:
According to the documents that I have, the 65816 CPU is an enhanced 16-bit version of the 65sc02, not the 65c02.  It doesn't have the `BBRn`, `BBSn`, `RMBn`, and `SMBn` instructions.

`$f7` is `"sbc [dp],y"` on that CPU.

Reply:
You're right, I'll need to fix that check, thank you for pointing it out.